mdn/developer-portal
# 1392: Add "related events" section to Topic pages, using banded-background styling

This changeset adds a new "related events" section to the Topic page, using the four-in-a-row component that we're adopting in a few other places.

It also adds support for automatically deciding whether certain components on the page should have a tinted background or not, in order to generate an alternate-banding look to the page. 

## Screenshots

(Some, but not all, of the permutations possible in the Topic page, showing the automatic tinted-backgrond logic working)

![Screenshot_2020-07-05 AV1 Video](https://user-images.githubusercontent.com/101457/86542131-ba09bd00-bf0a-11ea-9bd1-620de8891697.png)

-----

![Screenshot_2020-07-05 AV1 Video(1)](https://user-images.githubusercontent.com/101457/86542133-be35da80-bf0a-11ea-8cd7-361df58a71d1.png)

-----

![Screenshot_2020-07-05 AV1 Video(2)](https://user-images.githubusercontent.com/101457/86542134-bf670780-bf0a-11ea-8b41-35740d91f79f.png)

-----

![Screenshot_2020-07-05 AV1 Video(3)](https://user-images.githubusercontent.com/101457/86542135-c0983480-bf0a-11ea-9ada-4f0c51f940b0.png)

-----

![Screenshot_2020-07-05 AV1 Video(4)](https://user-images.githubusercontent.com/101457/86542136-c1c96180-bf0a-11ea-89ca-1606921bf02c.png)

Related issue: #1392

## Key changes:

- Modelling update to add Topic.related_events StreamField. This is so we can explicitly link to relevant Events, rather than using the Topic.events property, which gets the most recent Events, which isn't always what we want.
- Support manually specifying whether the four-card-row organism has a tinted background or not (and then ensuring it still does in the Events page's Featured panel)
- Support an optional tinted background on the People organism
- Add Topic.get_section_background_panel_hints helper to provide opinionated config to the template for whether a panel should or should not have a tinted background
- Support alternate tinted/not-tinted banding styling on Topics page. Uses the "tint hints" helper method in the Topic page template to pass configuration into the organisms it uses, if they are present

## How to test

- Code is on its way to staging
- Add some related events to a Topic page and preview/publish
- Play around with the varioous permutations of "Recent work", "Related events" and "Experts" sections being present (ie, having any content) and absent
- Confirm featured events styling (tint panel) is still correct and has a tint
- If there is a People section on the homepage, check that renders appropriately (no tint panel, because automatic tinting has not been applied to the homepage)
